velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -223,12 +256,28 @@ dwio::common::StripeProgress getStripeProgress(
  *
  * This method assumes each input `ColumnarBatch` have same schema.
  */
-void Writer::write(const VectorPtr& data) {
+void Writer::write(const VectorPtr& dataToWrite) {
+  VELOX_USER_CHECK(
+      dataToWrite->type()->equivalent(*schema_),
+      "The file schema type should be equal with the input rowvector type.");
+  auto rowVector =
+      std::dynamic_pointer_cast<facebook::velox::RowVector>(dataToWrite);
+  VELOX_CHECK_NULL(rowVector->nulls());
+  rowVector->setType(schema_);
+
+  auto data = std::make_shared<facebook::velox::RowVector>(
```

Comment:
Why do you need to make a new vector?

Reply:
@mbasmanova Removed.

Reply:
@JkSelf I'm not familiar with this code. At the high level, you need to start using schema_ instead of data->type() wherever you figure out the schema for the target file.

Reply:
@mbasmanova We used to create a new vector before in order to modify its type. However, now we can directly modify the type within the BaseVector, so there is no longer a need to create a new vector.